tests/Microsoft.Identity.Test.Unit/PublicApiTests/MtlsPopTests.cs

```diff
@@ -328,6 +328,74 @@ public async Task AcquireMtlsPopTokenForClientWithTenantId_SuccessAsync()
             }
         }
 
+        [TestMethod]
+        public async Task AcquireMtlsPopTokenForClientWithTenantIdCertChecks_Async()
```

Comment:
Do you have a test where you have a cached pop token associated with cert1 and then the callback returns cert2 ?

Reply:
yes we do, this one here - `PoP_CachedTokenWithDifferentCertificate_IsBypassedAsync`